packages/web-components/src/horizontalbarchart/horizontalbarchart.template.ts

```diff
@@ -14,9 +15,45 @@ getGroupper(tabsterCore);
  */
 export function horizontalbarchartTemplate<T extends HorizontalBarChart>(): ElementViewTemplate<T> {
   return html<T>` <template>
-    <div>
+    <div ${ref('rootDiv')} class="rootDiv">
       <div ${ref('chartContainer')}></div>
-      <div class="legendcontainer" ${ref('legendContainer')}></div>
+      ${when(
+        x => !x.hideLegends,
+        html<T>`
+          <div class="legendcontainer" role="listbox" aria-label="Legends">
+            ${repeat(
+              x => x.uniqueLegends,
+              html<ChartDataPoint, T>` <button
+                class="legend${(x, c) =>
+                  c.parent.activeLegend === '' || c.parent.activeLegend === x.legend ? '' : ' inactive'}"
+                role="option"
+                aria-setsize="${(x, c) => c.length}"
+                aria-posinset="${(x, c) => c.index + 1}"
+                aria-selected="${(x, c) => x.legend === c.parent.activeLegend}"
+                @mouseover="${(x, c) => c.parent.handleLegendMouseoverAndFocus(x.legend)}"
+                @mouseout="${(x, c) => c.parent.handleLegendMouseoutAndBlur()}"
+                @focus="${(x, c) => c.parent.handleLegendMouseoverAndFocus(x.legend)}"
+                @blur="${(x, c) => c.parent.handleLegendMouseoutAndBlur()}"
+                @click="${(x, c) => c.parent.handleLegendClick(x.legend)}"
+              >
+                <div class="legendRect" style="background-color: ${x => x.color}; border-color: ${x => x.color};"></div>
+                <div class="legendText">${x => x.legend}</div>
+              </button>`,
+            )}
+          </div>
+        `,
+      )}
+      ${when(
+        x => !x.hideTooltip && x.tooltipProps.isVisible,
+        html<T>`
+          <div class="tooltip" style="left: ${x => x.tooltipProps.xPos}px; top: ${x => x.tooltipProps.yPos}px">
+            <div class="tooltipline" style="border-left:4px solid ${x => x.tooltipProps.color};">
```

Comment:
the style can move to the class if it is static

Reply:
here, the color changes dynamically based on the element being hovered over or focused on